File: Add-Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string res = \\;
        int i = a.size() - 1, j = b.size() - 1, carry = 0;
        while (i >= 0 || j >= 0 || carry) {
            int x = i >= 0 ? a[i--] - '0' : 0;
            int y = j >= 0 ? b[j--] - '0' : 0;
            int sum = x + y + carry;
            res = char(sum % 2 + '0') + res;
            carry = sum / 2;
            }
            return res;
        }

};

File: Add-Binary.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuilder res = new StringBuilder();
        int i = a.length() - 1, j = b.length() - 1, carry = 0;
        while (i >= 0 || j >= 0 || carry == 1) {
            int x = i >= 0 ? a.charAt(i--) - '0' : 0;
            int y = j >= 0 ? b.charAt(j--) - '0' : 0;
            int sum = x + y + carry;
            res.insert(0, sum % 2);
            carry = sum / 2;
        }
        return res.toString();
    }

}